Add changelog for `8.x` (alpha) release (#3421)
getsentry/sentry-java: CHANGELOG.md

    - Current scope is forked often and data added to it is only added to events that are created while this scope is active. Data is also passed on to newly forked child scopes but not to parents.
- `Sentry.popScope` has been deprecated, please call `.close()` on the token returned by `Sentry.pushScope` instead or use it in a way described in more detail in "Migration Guide".
- We have chosen a default scope that is used for `Sentry.configureScope()` as well as API like `Sentry.setTag()`
    - For Android the type defaults to `CURRENT` scope
    - For Backend and other JVM applicatons it defaults to `ISOLATION` scope
- Event processors on `Scope` can now be ordered by overriding the `getOrder` method on implementations of `EventProcessor`. NOTE: This order only applies to event processors on `Scope` but not `SentryOptions` at the moment. Feel free to request this if you need it.

> - We are getting rid of `Hub` and replacing it with `Scopes`. We have deprecated and replaced some APIs. More details can be found in the "Migration Guide" section.

Reviewer: `nit`: How about phrasing it this way?
```suggestion
- `Hub` is deprecated in favor of `Scopes`, alongside some `Hub` relevant APIs. More details can be found in the "Migration Guide" section.
```
